lib/RefBackend/LowerToRefbackrtABI.cpp

```diff
@@ -38,7 +44,98 @@ static bool expressibleWithRefbackrtABI(FunctionType type) {
   // Currently, only memref types can be exposed at refbackrt ABI boundaries.
   return llvm::all_of(
       llvm::concat<const Type>(type.getInputs(), type.getResults()),
-      [](Type t) { return t.isa<MemRefType>(); });
+      [](Type t) { return t.isa<UnrankedMemRefType, MemRefType, FloatType>(); });
+}
+
+// Returns the integer rerpresentation of the CompilerDataStructures::ABIType
+// Must stay aligned with CompilerDataStructures::ABIArgType enum
+static uint32_t getIntReprForABIType(Type type) {
+  if (type.isa<MemRefType>() || type.isa<UnrankedMemRefType>()) {
+    return 1;
+  } else if (auto floatTy = type.dyn_cast<FloatType>()) {
+    switch (floatTy.getWidth()) {
+    case 32:
+      return 2;
+    case 64:
+      return 3;
+    default:
+      assert(false && "Unsupported float bit width");
+    }
+  } else if (auto intTy = type.dyn_cast<IntegerType>()) {
+  }
+  // assert(false && "couldn't get IntReprForABIType");
+  return -1;
+}
+
+// Must stay aligned with CompilerDataStructures::ABIElementType enum
+static uint32_t getIntReprForABIElementType(Type type) {
+  if (auto shapedTy = type.dyn_cast<ShapedType>()) {
+    auto elemTy = shapedTy.getElementType();
+    if (auto floatTy = elemTy.dyn_cast<FloatType>()) {
+      switch (floatTy.getWidth()) {
+      case 32:
+        return 1;
+      default:
+        assert(false && "Unsupported tensor element type");
+      }
+    }
+  }
+  return 0;
+}
+
+static SmallVector<int32_t, kMaxRank> getExtentsForType(Type type,
+                                                 const int32_t maxRank = kMaxRank) {
+  // Extend all shapes out to 4D to make our lives easier at the ABI boundary
+  if (auto shapedType = type.dyn_cast<ShapedType>()) {
+    if (!shapedType.hasRank()) {
+      // FIXME(brycearden): Not sure what we want to return here
+      return {1, 1, 1, 1, 1, 1};
```

Comment:
can you write this (and the other one below) in a way that uses kMaxRank? (so it is easy to change)